src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,98 +195,119 @@ public boolean canBeRotated() {
+            this.markForUpdate();
+        }
 
-                }
-                fuelTotal = fuelRemaining;
-                inventory.decrStackSize(4, 1);
-                this.markDirty();
-                this.markForUpdate();
-            }
+        if (ticksRemaining == 0 && inventory.getStackInSlot(0)==null && machineTier != MachineTier.TIER_0){
+            machineActive = false;
         }
 
-        if ((this.container.takePower(powerUsage, true) == powerUsage && machineTier != MachineTier.TIER_0) || fuelRemaining > 0) {
-            if (machineActive && !pulverizerPaused) {
+        if (this.canWork()){
+            if (machineTier == MachineTier.TIER_0){
+                fuelRemaining--;
+                machineActive = fuelRemaining > 0;
+            }else if (ticksRemaining > 0){
+                container.takePower(powerUsage,false);
+                machineActive = true;
+            }
+            if (machineActive && ticksRemaining > 0){
                 ticksRemaining--;
-
-                if (machineTier != MachineTier.TIER_0)
-                    this.container.takePower(powerUsage, false);
-                if (fuelRemaining > 0)
-                    fuelRemaining--;
-
             }
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-            if (inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
-                ItemStack itemIn = inventory.getStackInSlot(0);
-                ItemStack itemOut;
-
-                if (!PulverizerRegistry.containsInput(itemIn))
-                    return;
-
-                if (itemIn.stackSize - 1 <= 0) {
-                    itemOut = itemIn.copy();
-                    itemIn = null;
-                } else {
-                    itemOut = itemIn.copy();
-
-                    itemOut.stackSize = 1;
-                    itemIn.stackSize = itemIn.stackSize - 1;
-                }
-
-                if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
-                if (itemOut.stackSize == 0) itemOut = null;
+        if (fuelRemaining == 0 && machineTier == MachineTier.TIER_0 && inventory.getStackInSlot(4)!=null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0)
+        {
+            if (inventory.getStackInSlot(1)!= null) {
+                burnTime(1);
+            }
+            if (inventory.getStackInSlot(0)!= null){
+                burnTime(0);
+            }
+        }
 
-                inventory.setInventorySlotContents(0, itemIn);
-                inventory.setInventorySlotContents(1, itemOut);
 
-                ticksRemaining = 200;
-                machineActive = true;
 
-                this.markForUpdate();
-                this.markDirty();
-            }
+        if (this.canWork() && inventory.getStackInSlot(0)!=null && inventory.getStackInSlot(1)==null){
+            if (!cancrush(inventory.getStackInSlot(0)))
+                return;
 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00122)